app/controllers/users/sessions_controller.rb

```diff
@@ -3,13 +3,17 @@ class Users::SessionsController < Devise::SessionsController
   private
 
     def after_sign_in_path_for(resource)
-      if !verifying_via_email? && resource.show_welcome_screen?
+      if !verifying_via_email? && resource.show_welcome_screen? && !override?
```

Comment:
We don't want to skip the welcome page, 
it's nice to show it to all users the first time they login

What we want is to display the green check for all actions
Have a look at the gif "Initial welcome page when you sign up" [here](https://github.com/consul/consul/issues/2291)